frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -384,6 +385,86 @@ export function ConditionsTable(props: ConditionsTableProps) {
   );
 }
 
+interface VolumeMountsProps {
+  mounts:
+    | {
+        mountPath: string;
+        name: string;
+        readOnly: boolean;
+      }[]
+    | undefined;
+}
+
+function VolumeMounts(props: VolumeMountsProps) {
+  const { mounts } = props;
+  if (!mounts) {
+    return null;
+  }
+  const classes = MetadataDisplayStyle({});
+  return (
+    <Grid container direction="column" spacing={2}>
+      {mounts?.map((item, index) => (
+        <Grid item key={`${item.name}__${index}`}>
+          <Box display="flex">
+            <Typography className={classes.metadataValueLabel}>{item.mountPath}</Typography>
+            <Box px={2}>from</Box>
+            <Typography className={classes.metadataValueLabel}>{item.name}</Typography>
+            <Box px={1}>
+              <Typography className={classes.metadataValueLabel}>
+                {item.readOnly ? 'readOnly' : 'readWrite'}
+              </Typography>
+            </Box>
+          </Box>
+        </Grid>
+      ))}
+    </Grid>
+  );
+}
+
+function LivenessProbes(props: { liveness: KubeContainer['livenessProbe'] }) {
+  const classes = MetadataDisplayStyle({});
+
+  const { liveness } = props;
+
+  function LivenessProbeItem(props: { children: React.ReactNode; renderCondition: any }) {
+    const { renderCondition } = props;
+    return renderCondition ? (
+      <Box p={0.5}>
+        <Typography className={classes.metadataValueLabel} display="inline">
+          {props.children}
+        </Typography>
+      </Box>
+    ) : null;
+  }
+
+  return (
+    <Box display="flex" flexDirection="column">
+      <LivenessProbeItem renderCondition={liveness?.httpGet}>
+        {`http-get, path: ${liveness?.httpGet?.path}, port: ${liveness?.httpGet?.port}, 
+    scheme: ${liveness?.httpGet?.scheme}`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.exec}>
+        {`exec[${liveness?.exec?.command.join(' ')}]`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.successThreshold}>
+        {liveness?.successThreshold && `success = ${liveness?.successThreshold}`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.failureThreshold}>
+        {liveness?.failureThreshold && `failure = ${liveness?.failureThreshold}`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.initialDelaySeconds}>
+        {liveness?.initialDelaySeconds && `delay = ${liveness?.initialDelaySeconds}s`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.timeoutSeconds}>
+        {liveness?.timeoutSeconds && `timeout = ${liveness?.timeoutSeconds}s`}
+      </LivenessProbeItem>
+      <LivenessProbeItem renderCondition={liveness?.periodSeconds}>
+        {liveness?.periodSeconds && `period = ${liveness?.periodSeconds}s`}
+      </LivenessProbeItem>
```

Comment:
You are checking values when passing them to the `renderCondition` attr and also for creating their display.
I think you can remove the `renderCondition` attr altogether and just check if the children passed to it are valid. This will make the code easier to maintain.

Reply:
Fixed